### PR TITLE
AB#1938049 refresh node after value changed

### PR DESCRIPTION
**Pull request by ivywei0125**
What does this implement/fix? Explain your changes.
---------------------------------------------------
[AB#1938049](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1938049) refresh node after value changed

Does this close any currently open issues?
------------------------------------------

Any relevant logs, screenshots, error output, etc.?
-------------------------------------

Any other comments?
-------------------
…

Has this been tested?
---------------------------
- [ ] Tested

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-resource-connector-lib/src/main/java/com/microsoft/azure/toolkit/intellij/connector/ConnectorDialog.java [hunk 1]
```diff
@@ -124,6 +124,7 @@ protected void saveConnection(Connection<?, ?> connection) {
                     resource.getName(), consumer.getName());
                 AzureMessager.getMessager().success(message);
                 project.getMessageBus().syncPublisher(ConnectionTopics.CONNECTION_CHANGED).connectionChanged(project, connection, ConnectionTopics.Action.ADD);
+                project.getMessageBus().syncPublisher(ConnectionTopics.CONNECTIONS_REFRESHED).connectionsRefreshed();
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-resource-connector-lib/src/main/java/com/microsoft/azure/toolkit/intellij/connector/ConnectorDialog.java]: it think it's more suitable to emit this event in `connectionChanged`.

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-vm/src/main/java/com/microsoft/azure/toolkit/intellij/vm/creation/VMCreationDialog.java [hunk 2]
```diff
@@ -549,6 +556,27 @@ private AzureValidationInfo validateVirtualMachineName() {
...
+        // validate length
+        int nameMaxLen = 64;
+        if (cbImage.getSelectedItem() instanceof VmImage) {
+            nameMaxLen = ((VmImage) cbImage.getSelectedItem()).getOperatingSystem() == OperatingSystem.Windows ? 20 : 64;
+        }
+        if (StringUtils.isEmpty(name) || name.length() > nameMaxLen) {
+            return AzureValidationInfo.error(String.format(INVALID_LENGTH_MESSAGE, nameMaxLen), txtUserName);
+        }
+        // validate special character
+        if (!name.matches("^[A-Za-z-][A-Za-z0-9-_]*$")) {
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-vm/src/main/java/com/microsoft/azure/toolkit/intellij/vm/creation/VMCreationDialog.java]: make a static pre-compiled Regex instance may get （slightly） better performance.

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-vm/src/main/java/com/microsoft/azure/toolkit/intellij/vm/creation/VMCreationDialog.java [hunk 3]
```diff
@@ -81,6 +81,12 @@ public class VMCreationDialog extends AzureDialog<VirtualMachineDraft> implement
         "the multi-line PEM format. <p/> You can generate SSH keys using ssh-keygen on Linux and OS X, or PuTTYGen on Windows. </html>";
     public static final String SELECT_CERT_TITLE = "Select SSH public key";
     private static final String VIRTUAL_MACHINE_CREATION_DIALOG_TITLE = "Create Virtual Machine";
+    private static final String[] INVALID_USERNAMES = new String[]{"administrator", "admin", "user", "user1", "test",
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-vm/src/main/java/com/microsoft/azure/toolkit/intellij/vm/creation/VMCreationDialog.java]: should format the modified code before commit.

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-vm/src/main/java/com/microsoft/azure/toolkit/intellij/vm/creation/VMCreationDialog.java (hunk 2 above)
Comment (ivywei0125) [reply PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-vm/src/main/java/com/microsoft/azure/toolkit/intellij/vm/creation/VMCreationDialog.java]: use Pattern.compile instead

**Review by wangmingliang-ms — APPROVED** [review]
LGTM

**Pull request merged**